CorsixTH/Lua/hospitals/player_hospital.lua

```diff
@@ -216,6 +212,131 @@ function PlayerHospital:dailyAdviceChecks()
   end
 end
 
+--! Give advice about the player's progress towards winning or losing the level
+function PlayerHospital:giveLevelProgressAdvice()
+  -- Starting position
+  local level_config, level, start = self.world.map.level_config, self.world.map.level_number, {}
+  if level_config.towns and level_config.towns[level] then
+    start["balance"] = level_config.towns[level].StartCash
+    start["reputation"] = level_config.towns[level].StartRep
+  elseif level_config.town then
+    start["balance"] = level_config.town.StartCash
+    start["reputation"] = level_config.town.StartRep
+  end
+  if not start["balance"] then start["balance"] = 0 end
+  if not start["reputation"] then start["reputation"] = 500 end
+
+  -- Determine criteria of this level.
+  --  Some levels, including from the campaign, do not set every value
+  self.world:determineWinningConditions()
+  local criteria, msgs, criteria_count, criteria_met = self.world.goals, {}, 0, 0
+  if not criteria then return end
+  local win = {
+    reputation = criteria[1] and criteria[1]["win_value"],
+    balance = criteria[2] and criteria[2]["win_value"],
+    cured_p = criteria[3] and criteria[3]["win_value"], -- Patients cured percentage
+    cured_c = criteria[4] and criteria[4]["win_value"], -- Patients cured count
+    value = criteria[6] and criteria[6]["win_value"]
+  }
+  local lose = {
+    reputation = criteria[1] and criteria[1]["lose_value"],
+    balance = criteria[2] and criteria[2]["lose_value"],
+    killed_p = criteria[5] and criteria[5]["lose_value"] -- Patients killed percentage
+  }
+
+  -- Win criteria
+  --  1 Reputation
+  if win["reputation"] then
+    if self.reputation >= win["reputation"] then
+      if self.reputation > start["reputation"] then
+        table.insert(msgs, _A.level_progress.reputation_good_enough:format(win["reputation"]))
```

Comment:
Do I read correctly that you need to have (in this case) more reputation than than the stat as well as what the win criteria says?  If so, why not take `max(win["reputation"], start["reputation"])`  as win criteria?

Reply:
It felt strange to be told the hospital was good enough when I had lost reputation/balance since the level started. In the first level of the campaign, 400 is enough to win, but not enough to be congratulated.